shared/studio/tabs/auth/index.tsx

```diff
@@ -853,3 +901,28 @@ function Input({
     </div>
   );
 }
+
+function TextArea({
+  value,
+  onChange,
+  error,
+  size,
+}: {
+  value: string;
+  onChange: (val: string) => void;
+  error?: string | null;
+  size?: number;
+}) {
+  return (
+    <div className={styles.inputWrapper}>
+      <div className={cn(styles.input, {[styles.error]: !!error})}>
+        <textarea
+          value={value}
+          onChange={(e) => onChange(e.target.value)}
+          style={{height: 22 * (value?.split("\n").length ?? 1) + 10 + "px", width: size ? `${size}ch` : undefined}}
```

Comment:
I think the height needs to be something more like `32 * (value?.split("\n").length ?? 1) + 20 + "px"`, since the line height is 32px and at least on windows the scrollbars are closer to 20px.

Reply:
Also I think the textarea should have `whitespace: pre` and a `max-height` as currently the lines wrap, and it gets pretty tall if you're anywhere need the 128 line limit.

Reply:
> `whitespace: pre`

✅

> `max-height`

What do you think about a fixed height to cut down on the UI moving around as you're typing? Maybe just make it 3.5 * line-height tall?

Reply:
Here's `32 * 2.5` for with a fixed height, wdyt:

<img width="687" alt="image" src="https://github.com/edgedb/edgedb-ui/assets/1682194/3376ce37-ef92-46ee-b9a7-b78da9b8ca6e">

Reply:
Yeah, that also works. And user's can manually resize textarea's anyway.